src/lib.rs

```diff
@@ -63,7 +66,11 @@ pub trait SerializerAcceptor {
     /// The return type for the accept method
     type Output;
     /// Accept a serde::Serializer and do whatever you want with it.
-    fn accept<T: serde::Serializer>(self, T) -> Self::Output;
+    fn accept<T>(self, T) -> Self::Output
+    where
+        T: serde::Serializer,
+        T::Ok: Into<()>,
+        T::Error: Into<Error>;
```

Comment:
This looks way too generic to me. Does `T: Serializer<Ok = (), Error = Error>` work instead?

Reply:
Yeah, not sure what I was thinking.